### PR TITLE
FOUNDATIONS: CreateToken()

### PR DESCRIPTION
**Pull request by iamDiyorjon**

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        this.securityService = new SecurityService(
+            tokenBrokerMock.Object,
+            loggingBrokerMock.Object);
+    }
+
+    private string CreateRandomString() =>
+        new MnemonicString().GetValue();
+
+    private static User CreateRandomUser() =>
+            CreateUserFiller(dates: GetRandomDateTimeOffset()).Create();
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.cs:36]: Edit the tab

File: Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    private static DateTimeOffset GetRandomDateTimeOffset() =>
+      new DateTimeRange(earliestDate: DateTime.UnixEpoch).GetValue();
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.cs:39]: here too

File: Tarteeb.Api/Services/Foundations/SecurityService.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private readonly ILoggingBroker loggingBroker;
+
+    public SecurityService(ITokenBroker tokenBroker, ILoggingBroker loggingBroker)
+    {
+        this.tokenBroker = tokenBroker;
+        this.loggingBroker = loggingBroker;
+    }
+
+    public string CreateToken(User user) =>
+        tokenBroker.GenerateJWT(user);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Foundations/SecurityService.cs:24]: Bunda user validation mavjud emas

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

File: Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.Validations.Create.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            //then
+            actualUserValidationException.Should().BeEquivalentTo(
+                expectedUserValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+            broker.LogError(It.Is(SameExceptionAs(
+                expectedUserValidationException))), Times.Once);
+
+            this.tokenBrokerMock.Verify(broker =>
+            broker.GenerateJWT(It.IsAny<User>()), Times.Never);
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.Validations.Create.cs:41]: Fix tabs here

File: Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.Validations.Create.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+using Tarteeb.Api.Models.Users.Exceptions;
+using Xunit;
+
+namespace Tarteeb.Api.Tests.Unit.Services.Foundations
+{
+
+    public partial class SecurityServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnCreateJWTIfInputIsNullAndLogItAsync()
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/SecurityServiceTests.Validations.Create.cs:19]: instead of `async Task` use void

**Review by ElbekDeveloper — CHANGES_REQUESTED** [review]

**Review by ElbekDeveloper — APPROVED** [review]

**Pull request merged**